### ISSUE TITLE
Enum and no-shadow - False Positive

### ISSUE THREAD

**Issue by ro-savage** [eslint/eslint#13640]
* **ESLint Version:** v7.7.0
* **Node Version:** v14.4.0
* **yarn Version:** 1.22.4

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
"typescript-eslint-parser": "^22.0.0",

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": [
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/recommended-requiring-type-checking",
        "plugin:jest/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "project": "./tsconfig.json",
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint",
        "react-hooks",
        "jest"
    ],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
enum GraphType {
  Availability = "availability",
  Capacity = "capacity",
  Utilization = "utilization",
  Workload = "workload",
  ProjectBillings = "project-billings",
  PeopleCost = "people-costs",
  ProjectProfit = "project-profits",
}

export default GraphType;

```

```bash
eslint 'app/javascript/**/*.{ts,tsx}
```

**What did you expect to happen?**
Eslint is incorrectly showing a no-shadow error for enums. When there is no our declarations in files.

**What actually happened? Please include the actual, raw output from ESLint.**
`1:6  error  'GraphType' is already declared in the upper scope  no-shadow`
![image](https://user-images.githubusercontent.com/9244507/91770527-69c28900-ec35-11ea-9bcd-d5215d39efca.png)

**Comment by kaicataldo** [comment]
Since this is related to TypeScript, chances are we won't be able to fix this on our end. Have you checked to see if there's a corresponding rule in `@typescript-eslint/plugin`?

**Comment by ro-savage** [comment]
Thanks for the amazing response time @kaicataldo - turning off `no-shadow` in eslint, and turning it on for `@typescript-eslint/plugin` fixed the issue.
